### PR TITLE
docs: Fix backticks in api protos

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Albere <alberenorthey@gmail.com>

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/type/matcher/regex.proto [hunk 1]
```diff
@@ -21,11 +21,11 @@ message RegexMatcher {
   // the documented `syntax <https://github.com/google/re2/wiki/Syntax>`_. The engine is designed
   // to complete execution in linear time as well as limit the amount of memory used.
   //
-  // Envoy supports program size checking via runtime. The runtime keys `re2.max_program_size.error_level`
-  // and `re2.max_program_size.warn_level` can be set to integers as the maximum program size or
+  // Envoy supports program size checking via runtime. The runtime keys ``re2.max_program_size.error_level``
+  // and ``re2.max_program_size.warn_level`` can be set to integers as the maximum program size or
```
Comment (phlax) [inline api/envoy/type/matcher/regex.proto:25]: not sure if these are necessary to fix v3 protos - i think the v3 protos are getting this from xds

**Comment by phlax** [comment]
cc @albere

File: api/envoy/type/matcher/regex.proto (hunk 1 above)
Comment (htuch) [reply api/envoy/type/matcher/regex.proto:25]: Yes, these are v2 protos.

**Review by htuch — APPROVED** [review]

**Comment by htuch** [comment]
@phlax looking at a reasonable samples this LGTM. Do you have some systematic way to avoid us getting into this pitfall? Can we ban single ticks entirely and use other ways to achieve the same format effect?

**Comment by phlax** [comment]
> Do you have some systematic way to avoid us getting into this pitfall? Can we ban single ticks entirely and use other ways to achieve the same format effect?

yes, and yes

we currently have the `Bad RST` warning in `protodoc` the intention is to turn that to error (perhaps with a `--continue` option)

**Pull request merged**